webapp/store/selectors/nav.js

```diff
@@ -0,0 +1,176 @@
+import { plugins } from '@bpanel/bpanel-utils';
+
+const { comparePlugins } = plugins;
+
+const getMetadataList = state => Object.values(state.pluginMetadata);
+const getSidebarItems = state => state.nav.sidebar;
+
+/* Sort plugin metadata (by order and name) and insert and
+ * sort children after parents
+ * @param {Array{}} pluginMeta - unsorted array of plugin Metadata
+ * @returns {Array{}} plugins - returns an array of sorted metadata objects
+ */
+export const sortPluginMetadata = (pluginMeta = []) => {
+  assert(Array.isArray(pluginMeta), 'Must pass array of metadata');
+  const subItems = new Map();
+
+  const sortedPluginMetadata = pluginMeta
+    .filter(plugin => {
+      if (plugin.parent) {
+        // if it has a parent then add it to corresponding array in subItems map
+        // this is so that we can get all subItems and then sort later
+        const { parent } = plugin;
+        if (!subItems.has(parent)) {
+          subItems.set(parent, [plugin]);
+        } else {
+          const children = subItems.get(parent);
+          children.push(plugin);
+          subItems.set(parent, children);
+        }
+        // don't return to array of top level items
+        return false;
+      } else {
+        // only returning an array of the parents
+        return true;
+      }
+    })
+    .sort(comparePlugins);
+
+  subItems.forEach((children, parent) => {
+    const index = sortedPluginMetadata.findIndex(
+      plugin => parent === plugin.name
+    );
+    children.sort(comparePlugins);
+    sortedPluginMetadata.splice(index + 1, 0, ...children);
+  });
+
+  return sortedPluginMetadata;
+};
+
+/* Update list of plugin metadata (or nav objects) to have properly
+ * formatted, nested paths
+ * e.g. { name: 'child', parent: 'parent' } becomes
+ * { name: 'child', parent: 'parent', pathName: 'parent/child'}
+ * @param {Array{}} metadata - Array of plugin metadata objects
+ * @returns {Array{}} plugins - returns an array of metadata w/ updated paths
+ */
+export function getNestedPaths(metadata) {
+  assert(Array.isArray(metadata), 'Must pass array of metadata');
+  return metadata.reduce((updated, plugin) => {
+    if (plugin.parent) {
+      // if this plugin is a child then update its pathName property
+      // to nest behind the parent
+      const parentIndex = metadata.findIndex(
+        item => item.name === plugin.parent
+      );
+
+      const parent = metadata[parentIndex];
+      // eslint-disable-next-line no-console
+      console.assert(
+        parent,
+        `Parent ${plugin.parent} could not be found for child ${plugin.name}`
+      );
+      // get parent path, if none, use name
+      const parentPath = parent.pathName ? parent.pathName : parent.name;
+
+      // get plugin path, if none, use name
+      let pathName = plugin.pathName ? plugin.pathName : plugin.name;
+
+      // sanitize of any other uri constructions like other nested paths
+      pathName = pathName.slice(pathName.lastIndexOf('/') + 1);
```

Comment:
What if `pathName` is `/`?

Reply:
The function `sanitizePaths` above now will remove any `/` at the beginning of a path. So when you get here, the pathName will be empty and will get set to the name instead now.

Reply:
👍